include/extractor/guidance/intersection_generator.hpp

```diff
@@ -22,6 +22,13 @@ namespace extractor
 {
 namespace guidance
 {
+
+struct IntersectionGenerationParameters
+{
+    NodeID nid;
+    EdgeID via_eid;
+};
```

Comment:
Hm. I don't feel that it is reasonable to create an intermediate struct every time to call a handler. Whats your take?